File: 1469-minimum-number-of-steps-to-make-two-strings-anagram/1469-minimum-number-of-steps-to-make-two-strings-anagram.cpp
class Solution {
public:
    int minSteps(string s, string t) {
        int* dmap = new int[256]{};
        for (const auto c: s) {
            ++dmap[c];
        }
        for (const auto c: t) {
            --dmap[c];
        }
        int steps = 0;
        for (int i = 0; i < 256; ++i) {
            const int diff = dmap[i];
            steps += diff > 0 ? diff : 0;
        }
        
        delete[] dmap;
        return steps;
    }
};

File: 1469-minimum-number-of-steps-to-make-two-strings-anagram/1469-minimum-number-of-steps-to-make-two-strings-anagram.rs
impl Solution {
    pub fn min_steps(s: String, t: String) -> i32 {
        let mut dmap = [0;26];
        
        for c in s.chars() {
            dmap[c as usize - 'a' as usize] += 1;
        }
        
        for c in t.chars() {
            dmap[c as usize - 'a' as usize] -= 1;
        }

        dmap.into_iter().filter(|i| **i > 0).sum()
    }
}